File: app/src/main/java/io/github/ovso/righttoknow/framework/ad/MyAdView.java
package io.github.ovso.righttoknow.framework.ad;

import android.content.Context;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdSize;
import com.google.android.gms.ads.AdView;
import io.github.ovso.righttoknow.Security;

public class MyAdView {

  public static AdView getAdmobAdView(Context context) {
    AdView adView = new AdView(context);
    adView.setAdSize(AdSize.SMART_BANNER);
    adView.setAdUnitId(Security.ADMOB_UNIT_ID.getValue());
    AdRequest adRequest = new AdRequest.Builder().build();
    adView.loadAd(adRequest);
    return adView;
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/framework/ad/MyAdView.kt
package io.github.ovso.righttoknow.framework.ad

import android.content.Context
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdSize
import com.google.android.gms.ads.AdView
import com.google.android.gms.ads.InterstitialAd
import io.github.ovso.righttoknow.Security

object MyAdView {
  fun getAdmobAdView(context: Context?): AdView {
    val adView = AdView(context)
    adView.adSize = AdSize.SMART_BANNER
    adView.adUnitId = Security.ADMOB_UNIT_ID.getValue()
    val adRequest = AdRequest.Builder().build()
    adView.loadAd(adRequest)
    return adView
  }

  @JvmStatic
  fun getInterstitalAd(context: Context?): InterstitialAd {
    val interstitialAd = InterstitialAd(context)
    interstitialAd.adUnitId = Security.ADMOB_INTERSTITIAL_UNIT_ID.getValue()
    val adRequestBuilder = AdRequest.Builder()
    interstitialAd.loadAd(adRequestBuilder.build())
    return interstitialAd
  }
}